src/server/simulator.js

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+var Q = require('q'),
+    fs = require('fs'),
+    path = require('path'),
+    log = require('./utils/log'),
+    dirs = require('./dirs'),
+    utils = require('./utils/jsUtils'),
+    Configuration = require('./config'),
+    Project = require('./project'),
+    SimulationServer = require('./server'),
+    Telemetry = require('./telemetry');
+
+/**
+ * @param {object} opts Configuration for the current simulation.
+ * @constructor
+ */
+function Simulator(opts) {
+    this._config = this._parseOptions(opts);
+    this._state = Simulator.State.IDLE;
+    this._urls = null;
+
+    this.hostRoot = {
+        'app-host':  path.join(dirs.root, 'app-host')
+    };
+
+    var simHostOptions = this._config.simHostOptions;
+    Object.defineProperty(this.hostRoot, 'sim-host', {
+        get: function () {
+            // Get dynamically so simHostOptions is initialized
+            return simHostOptions.simHostRoot;
+        }
+    });
+
+    var platform = opts.platform || 'browser';
+
+    this._telemetry = new Telemetry(this, this._config.telemetry);
+    this._project = new Project(this, platform);
+    this._server = new SimulationServer(this);
```

Comment:
I have a proposal for decoupling this, instead of passing through parameters `this`, and generating a circular dependency, I can create a kind of proxy object that contains the API that we want to expose for the instances, and that is a subset of all of the APIs. Doing this will help us too to testing the objects, without the need of a full mock of the main instance. I will give a try and then share to you for your comments :)